src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -315,20 +439,27 @@ and not exists (select prs from participation.results prs
      * @param teamShortName the short name of the team for which to consider participations
      * @return Tuples of participation ids and number of submissions per participation
      */
-    @Query("select participation.id, count(submissions) from StudentParticipation participation left join participation.submissions submissions where participation.team.shortName = :#{#teamShortName} and participation.exercise.course.id = :#{#courseId} group by participation.id")
+    @Query("""
+            SELECT p.id, COUNT(s) FROM StudentParticipation p
+            LEFT JOIN p.submissions s
+                WHERE p.team.shortName = :#{#teamShortName}
+                AND p.exercise.course.id = :#{#courseId}
+            GROUP BY p.id
+            """)
     List<long[]> countSubmissionsPerParticipationByCourseIdAndTeamShortName(@Param("courseId") long courseId, @Param("teamShortName") String teamShortName);
 
+    // TODO SE Improve!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! // maybe leave out max id line
```

Comment:
This TODO is for a follow-up then?

Reply:
yep most likely, I removed some of the exclamation marks